desktop/plugins/network/ManageMockResponsePanel.tsx

```diff
@@ -185,7 +198,7 @@ export function ManageMockResponsePanel(props: Props) {
     }
     setCurrentRouteSize(routeSize);
   }, [routes]);
-  const duplicatedIds = useMemo(() => _duplicateIds(routes), [routes]);
+  const duplicatedIds = useMemo(() => _duplicateIds(props.routes), [routes]);
```

Comment:
Just one nit here: Shouldn't the dependency here also change from `routes` to `props.routes` to ensure this does update correctly?

Reply:
I think you're right, I'll make that change.  Thanks.

Reply:
@passy. I made the fix but it caused a crash in testing so I reversed it.

Reply:
It appears to be checking (and memorizing) the list of duplicate ids.  Which seems to be working fine.  

"routes" is assigned from "props.routes" in line 189.

Reply:
@passy Looks like there is a problem with routes created through "copy highlighted".  I'm now starting to get an error message that occurs when I update a route that was copied.  It looks to be in an NDK library on the device (but that's just my guess).  Here's the crash report:

![image](https://user-images.githubusercontent.com/337874/90057372-ee109480-dca5-11ea-91a1-eb523042bcca.png)

Reply:
@passy   I have not been able to reproduce the error.  For now, I'm assuming it was just a problem for me.

However, while researching empty string problems (which is what the stack trace seems to suggest is the issue) I did find a related edge case.  How should an empty value for an HTTP header be handled?

The Flipper plugin was not adding the header if either the name or value was empty.  But according to the HTTP specification, an empty value is allowed:

[](https://stackoverflow.com/questions/12130910/how-to-interpret-empty-http-accept-header)

Reply:
@passy. Some additional info on your comment.

You asked about this line:

  const duplicatedIds = useMemo(() => _duplicateIds(props.routes), [routes]);

You  asked if the 2nd occurence of "routes" should also be "props.routes".  The short answer is yes and I did make the change.

After consulting with someone who knows react, they pointed out that by changing the Props object, the deconstruction of routes no longer works the same, so I had to remove this line as well:

    const {routes} = props;

Now everything seems to work properly.

Also, was wondering how you added the code to your comment.  I'd like to be able to do that also.

Reply:
@bizzguy I might be missing something, but things should behave exactly the same when and when not destructuring `routes`, as `props` is a new object every render. If it does matter, it sounds like there is something side-effectful happening during render? So this render component should behave exactly the same with and without this change, and if there is a change, I'd be a little bit concerned :)